app/notes/[noteId].tsx

```diff
@@ -0,0 +1,74 @@
+import { useFocusEffect, useLocalSearchParams } from 'expo-router';
+import { useCallback, useState } from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+import { colors } from '../../constants/colors';
+import type { NoteResponseBodyGet } from '../api/[noteId]+api';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    alignItems: 'center',
+    justifyContent: 'center',
+    backgroundColor: colors.background,
+    padding: 10,
+  },
+  textContainer: {
+    alignItems: 'center',
+    gap: 12,
+  },
+  title: {
+    fontSize: 24,
+    fontFamily: 'Poppins_700Bold',
+    color: colors.text,
+    textAlign: 'center',
+  },
+  textContent: {
+    fontSize: 16,
+    fontFamily: 'Poppins_400Regular',
+    color: colors.textSecondary,
+    textAlign: 'center',
+    marginTop: 8,
+  },
+});
+
+export default function NotePage() {
+  const { noteId } = useLocalSearchParams();
+
+  const [title, setTitle] = useState('');
+  const [textContent, setTextContent] = useState('');
+
+  useFocusEffect(
+    useCallback(() => {
+      async function loadNote() {
+        if (typeof noteId !== 'string') {
+          return;
+        }
+
+        const response = await fetch(`/api/${noteId}`);
+        const body: NoteResponseBodyGet = await response.json();
+
+        if ('note' in body) {
+          setTitle(body.note.title);
+          setTextContent(body.note.textContent);
+        }
+      }
+
+      loadNote().catch((error) => {
+        console.error(error);
+      });
+    }, [noteId]),
+  );
+
+  if (typeof noteId !== 'string') {
+    return null;
+  }
```

Comment:
There should be some kind of "Access Denied" message, similar to the Next.js example and the GraphQL example:

- https://github.com/upleveled/next-js-example-spring-2024-atvie/blob/8dbf5ccc628b57056296ff05de97fba124699503/app/notes/%5BnoteId%5D/page.tsx#L23-L32

Reply:
https://github.com/upleveled/expo-example-fall-2024-atvie/pull/1/commits/3a7495cd93fe12b0633e56e7b6048d097ef216dd